addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +115,128 @@ def _get_duration_from_tracking(self, trackings):
+      - _resource_is_not_rotting_hook(task) must be overriden to add additional conditions for which a stage IS NOT rotting
+        (e.g. a task that has been closed):
+            def _resource_is_not_rotting_hook(self, task):
+                if task.is_closed:
+                    return True
+                return super()._resource_is_not_rotting_hook(task)
+      - _compute_is_rotting() must be overriden to update its @depends to trigger if the "days to rot" field is updated,
+        or if the variables on which _resource_is_not_rotting_hook() override depends are modified:
+            @api.depends('is_closed', 'stage_id.day_rot')
+            def _compute_rotting(self):
+                super()._compute_rotting()
+      - _search_is_rotting() should be overriden to update the returned domain with additional conditions for which a stage COULD BE rotting
+        (AND condition, all need to be true):
+            def _search_is_rotting(self, operator, value):
+                sup = super()._search_is_rotting(operator, value)
+                return Domain.AND([sup, [('is_closed', '=', True)]])
+
+    - The is_rotting, day_rotting, last_activity fields need to be added to the relevant views
+        (as well as the "days to rot" field on the tracked model).
+        You may want to use the rotting_form and rotting_kanban field widgets to display the fields visually on form and kanban view
+        (please note- these widgets need both day_rotting and is_rotting fields on the view to function).
+
+
+    Note that if _stage_day_rot_field is not set, or if the value stored by the field pointed by _stage_day_rot_field is 0,
+    then the resource will never rot.
+    """
+
+    is_rotting = fields.Boolean('Rotting', compute='_compute_rotting', search='_search_is_rotting')
+    day_rotting = fields.Integer('Days Rotting', help='Day count since this resource was last updated',
+        compute='_compute_rotting')
+    last_activity = fields.Date('Date of last activity', compute="_compute_last_activity", store=True, readonly=False)
+
+    @api.depends('write_date')
+    def _compute_last_activity(self):
+        for resource in self:
+            resource.last_activity = resource.write_date or self.env.cr.now()
+
+    def _get_day_count_to_rotting(self) -> int:
+        """
+        :return: day count before the resource is considered to be rotting
+        """
+        self.ensure_one()
+        rotting_stage = self[self._track_duration_field]
+        if not rotting_stage or not hasattr(self, '_stage_day_rot_field'):
+            # If _stage_day_rot_field has not been set, the rotting feature is not enabled for this model
+            return 0
+        if not self._stage_day_rot_field in rotting_stage:
+            raise UserError(_('Models using the rotting feature need to declare a "day_rot" field on their stage model. Please refer to the help present in the mail/models/mail_tracking_duration_mixin.py file for implementation details'))
```

Comment:
UserErrors are meant for end users which would have 0 clue about what to do with this :smile: 

Would say either do nothing when misconfigured, either raise a ValueError with a non-translated message meant for developers.